pkg/pillar/cmd/baseosmgr/handlebaseos.go

```diff
@@ -259,6 +259,33 @@ func doBaseOsActivate(ctx *baseOsMgrContext, uuidStr string,
 
 	log.Functionf("doBaseOsActivate: %s activating", uuidStr)
 
+	// Before writing to partition lets make sure we have enough space on the partition
+	// 1. Get the size of the image using contenttree status
+	// 2. Get the size of the partition using zboot
+	// 3. If partition size is < image size, error out
+
+	cts := lookupContentTreeStatus(ctx, status.ContentTreeUUID)
+
+	if cts == nil {
+		errString := fmt.Sprintf("doBaseOsActivate: ContentTreeStatus not found for %s", status.ContentTreeUUID)
+		log.Error(errString)
+		status.SetErrorNow(errString)
+		changed = true
+		return changed
+	}
+
+	if cts.State == types.LOADED {
+		partSize := zboot.GetPartitionSizeInBytes(status.PartitionLabel)
+		imageSize := cts.MaxDownloadSize
```

Comment:
Don't we have an actual size which could be smaller than MaxDownloadSize?

Reply:
MaxDownloadSize seems to be the size of image sent from the controller in the content tree config. zedagent is parsing that value from the config. To me that seems to be the actual size of the image. Let me know if that is not correct.